### DISCUSSION TITLE
Allow tls on MQTT Adapters

### DISCUSSION THREAD

**Discussion by heinehenning** [apache/streampipes#1528] [Ideas]
When creating an MQTT adapter, there is no option to use tls encryption. Did I miss that or could that be a possible feature?

_A little disclaimer about me: This is my first day on githup and my first day using streampipes.
I first heard about streampipes during buildingIOT in munich. From what I have seen so far it is a great tool and I would love to see it thrive!_

**Comment by dominikriemer** [comment]
Hi @heinehenning and welcome to StreamPipes! (and Github :-))

First of all, thanks for starting the discussion!
TLS encryption is currently not supported by the adapter but would be a very useful addition. The MQTT sink already supports more configuration option and it should be easy to add these to the adapter as well.

I'll create an issue for that - if you think other config options would be useful to have, just comment in this thread ;-)

**Reply by bossenti** [reply]
Issue is created: https://github.com/apache/streampipes/issues/1530

**Comment by bossenti** [comment]
Thank you for contributing your ideas @heinehenning 🎉
